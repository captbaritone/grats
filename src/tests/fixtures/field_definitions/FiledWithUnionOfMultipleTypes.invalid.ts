/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello(): string | boolean {
    return "Hello world!";
  }
}
