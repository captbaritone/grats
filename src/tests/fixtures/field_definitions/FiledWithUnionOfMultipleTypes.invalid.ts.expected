-----------------
INPUT
----------------- 
/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello(): string | boolean {
    return "Hello world!";
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/field_definitions/FiledWithUnionOfMultipleTypes.invalid.ts:4:12 - error: Expected exactly one non-nullish type. GraphQL does not support fields returning an arbitrary union of types. Consider defining an explicit `@gqlUnion` union type and returning that.

4   hello(): string | boolean {
             ~~~~~~

  src/tests/fixtures/field_definitions/FiledWithUnionOfMultipleTypes.invalid.ts:4:21
    4   hello(): string | boolean {
                          ~~~~~~~
    Other non-nullish type
